src/Raven.Server/Documents/Sharding/Handlers/ReshardingHandler.cs

```diff
@@ -63,7 +64,10 @@ public async Task StartResharding()
                             if (bucket > toBucket)
                                 break;
 
-                            await ServerStore.Sharding.StartBucketMigration(database, bucket, toShard, $"{raftId ?? Guid.NewGuid().ToString()}/{bucket}");
+                            var (index, _) = await ServerStore.Sharding.StartBucketMigration(database, bucket, toShard,
+                                $"{raftId ?? Guid.NewGuid().ToString()}/{bucket}");
+                            await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, index, token.Token);
```

Comment:
What if there is a failure? 
You are also not actually waiting for the result, are you?

Reply:
No, the only thing I care here - that it is committed, so the `HasActiveMigrations` above will not be racy.
It is a user operation, so on error, we will report that to the user (studio) and stop the process.